encryptio.py

```diff
@@ -1,55 +1,302 @@
+        case CryptoModes.EAX:
+            mode = Blowfish.MODE_EAX
+        case CryptoModes.ECB:
+            mode = Blowfish.MODE_ECB
+            data = pad(data, Blowfish.block_size)
+        case CryptoModes.OFB:
+            mode = Blowfish.MODE_OFB
+        case _:
+            print('This mode is not available for the selected algorithm')
+            return
+
+    if mode != Blowfish.MODE_CTR:
+        cipher = Blowfish.new(key, mode)
+
+    if mode == Blowfish.MODE_EAX:
+        ciphertext, tag = cipher.encrypt_and_digest(data)
+    else:
+        ciphertext = cipher.encrypt(data)
+
+    if mode == Blowfish.MODE_EAX:
+        additional_information = ['nonce_tag', cipher.nonce + tag]
+    elif mode == AES.MODE_ECB:
+        additional_information = []
+    else:
+        additional_information = ['initialization_vector', cipher.iv]
+
+    create_binary_file(ciphertext, 'BLOWFISH', additional_information)
+
+
+## mode execution
+def encrypt_mode(args):
+    match args.cryptoAlgorithm:
+        case CryptoAlgorithms.AES:
+            encrypt_aes(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode)
+        case CryptoAlgorithms.DES:
+            encrypt_des(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode, DES)
+        case CryptoAlgorithms.TRIPLEDES:
+            encrypt_des(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode, DES3)
+        case CryptoAlgorithms.IDEA:
+            encrypt_idea(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode)
+        case CryptoAlgorithms.BLOWFISH:
+            encrypt_blowfish(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode)
+
+
+# def decrypt_mode(args):
+
+# def bruteforce_mode(args):
+
+## mode listing
+def arg_encryption_mode(args):
+    encrypt_mode(args)
+
+
+def arg_decryption_mode(args):
+    encrypt_mode(args)
+
+
+# def arg_bruteforce_mode(args):
+#     bruteforce_mode(args)
+
+
 ## argument checker
 def check_key(key):
-
     if not (key.isalpha()):
         raise argparse.ArgumentTypeError('Key is not a string')
 
-    if(len(key) < 8): 
+    if len(key) < 8:
         raise argparse.ArgumentTypeError('Key to short - Minimum 8 characters')
 
     return key
-    
-## parse arguments 
-parser = argparse.ArgumentParser(description='encryptio.py is an easy to use generic encryption tool', epilog='--- encryptio.py - Moritz Nentwig Laura Tzigiannis', add_help=True)
 
-parser.add_argument('encryptionAlgorithm', help='Algorithm to encrypt', type=EncryptionAlgorithms, choices=EncryptionAlgorithms)
-parser.add_argument('encryptionKey', help='Key to encrypt with', type=check_key)
-parser.add_argument('encryptionLibrary', help='library to encrypt object with', type=str)
 
-parser.add_argument('-o', '--outputFile', help='file to write output to', type=argparse.FileType('w'), dest='outputFile')
+def create_binary_file(ciphertext, algorithm, additional_information):
```

Comment:
A Helpful Comment above the method name should provide information about what is done in the method. Have a look at the other methods.

Reply:
Done.